controller/src/beerocks/master/db/db.cpp

```diff
@@ -2359,6 +2364,12 @@ bool db::is_channel_in_pool(const sMacAddr &mac, uint8_t channel, bool single_sc
 
     auto &pool =
         (single_scan ? hostap->single_scan_config : hostap->continuous_scan_config).channel_pool;
+
+    if (pool.size() == 1 && *pool.begin() == SCAN_ALL_CHANNELS) {
```

Comment:
scan all channels should set the pool to empty and set a member bool scan_all_channels=true instead of this hack.

Reply:
See my comment below